commit 188a8a5a7aa66ea48af96cda70bac300bb8bc9bc
Author: Sjoerd Simons <sjoerd@luon.net>
Date:   Wed Oct 18 23:44:35 2006 +0200

    use $HAL_PROP_POWER_MANAGEMENT_TYPE to check the power management type
    
    HAL_PROP_LAPTOP_PANEL_ACCESS_METHOD was used to check if the
    powermanagement type, which isn't set on the root object. Using
    HAL_PROP_POWER_MANAGEMENT_TYPE fixes suspend on pmu machines

diff --git a/tools/linux/hal-system-power-suspend-linux b/tools/linux/hal-system-power-suspend-linux
index 6842cc0..dad5198 100755
--- a/tools/linux/hal-system-power-suspend-linux
+++ b/tools/linux/hal-system-power-suspend-linux
@@ -19,7 +19,7 @@ unsupported() {
 read seconds_to_sleep
 
 #PMU systems cannot use /sys/power/state yet, so use a helper to issue an ioctl
-if [ "$HAL_PROP_LAPTOP_PANEL_ACCESS_METHOD" = "pmu" ]; then
+if [ "$HAL_PROP_POWER_MANAGEMENT_TYPE" = "pmu" ]; then
 	hal-system-power-pmu sleep
 	if [ $? -ne 0 ]; then
 		echo "org.freedesktop.Hal.Device.SystemPowerManagement.NotSupported" >&2
